pkg/assembler/backends/inmem/certifyVuln.go

```diff
@@ -165,7 +165,7 @@ func (c *demoClient) ingestVulnerability(ctx context.Context, packageArg model.P
 		if noKnownVulnID != 0 && noKnownVulnID == v.noKnownVulnID {
 			vulnMatch = true
 		}
-		if vulnMatch && packageID == v.packageID && certifyVuln.TimeScanned.UTC() == v.timeScanned && certifyVuln.DbURI == v.dbURI &&
+		if vulnMatch && packageID == v.packageID && certifyVuln.TimeScanned.Equal(v.timeScanned) && certifyVuln.DbURI == v.dbURI &&
```

Comment:
hmmm..I don't understand why there are duplicate nodes in the first place. No matter what the timestamp says, there should just be one `certifyVuln` node per package version. The OSV certifier has the ability to check the timestamp and create a new `certifyVuln` node if that is out of date (based on user input), but that is disabled by default.

Reply:
@pxp928 As far as I saw, osv certifier uses `PackageNode`s which have multiple instances for a purl (with different hashes). I'm not sure if this is correct or not, but it ends up in trying to ingest vuln multiple times. I think there's more room in improving this flow.

Reply:
So it checks at the package version level (which should be unique). So each package version should only have one `certifyVuln` and not be ingested multiple times.

Reply:
@pxp928 I think the issue originates somewhere here

https://github.com/guacsec/guac/blob/main/pkg/certifier/osv/osv.go#L84

As there could be multiple nodes for the purl, so the certifier will create multiple documents. I can dig deeper into this next week.

Reply:
ah yes I see, we could remove the need to do it for each hash. As it's not needed.